docs/runtime_image.md

```diff
@@ -0,0 +1,83 @@
+# Runtime image
+
+## Overview
+
+For a dynamic languages like PHP, Python or Ruby, the build-time and run-time environments are the same. In this case using builder as a base image for a resulting image is natural.
+
+For compiled languages like C/C++, Java or Golang, the dependencies necessary for compilation might dramatically outweigh the actual runtime artifacts. To keep runtime images slim, S2I enables multiple-step build processes, where a binary artifact such as an executable or Java WAR file is created in the first builder image, extracted, and injected into an second image that simply places the executable in the correct location for execution. To give you even more abilities to customize a resulting image, S2I is also executing `assemble-runtime` script inside of this (runtime) image. In this way you may do a final adjustments by modifying files before an image will be committed.
+
+The following scheme illustrates the build workflow:
+
+![s2i workflow](./runtime-image-flow.png "s2i workflow")
+
+
+## How it works?
+
+To make this work S2I needs to know the following:
+
+* builder image
+* mapping of the artifacts
+* runtime image
+
+And all these information you should specify during S2I invocation:
+
+    s2i build <repo> <builder-image> <my-app> --runtime-image <runtime-image> --runtime-artifact </path/to/artifact>
+
+The only new here are `--runtime-image` and `--runtime-artifact` options. The latter option is used to specify a full path to file (or directory) inside of the builder image and a corresponding directory on the runtime image where S2I will copy these files. Note that you can specify this option multiple times.
```

Comment:
> In the "How it works" chapter I tried to hide some details to show a full picture. And that's why I'm not talking about optional parameter of the --runtime-artifact option here.

that's fine but then you should fix the sentence that says "The latter option is used to specify a full path to file (or directory) inside of the builder image and a corresponding directory on the runtime image where S2I will copy these files." to just say "The latter option is used to specify a full path to file (or directory) inside of the builder image that will be copied into the `WORKDIR` of the runtime image"

because right now you're referring to a "corresponding directory" that you aren't actually specifying in that example, so it's confusing.

> Would you mind to leave this explanation in the Format and behavior of the --runtime-artifact option chapter? It's very rare case as I think and (probably) beginners won't use it at all.

yeah I hadn't gotten to your more detailed section when I made those comments...  the detailed section definitely helps provide the needed specifics.

> (Moreover, I started to think that we can remove it. If user wants to move the files, he can do it in the assemble-runtime script. So, less code, less misunderstanding and optional parameters, less documentation but user still will be able to achieve his goal. WDYT?)

I think we still need it.  I don't want to have to override the assemble-runtime script just because my maven build produces a file named "foo.war" and my assemble-runtime script only looks for "ROOT.war" or something.  (There are probably better examples).